includes/notes/class-wc-stripe-upe-availability-note.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Display a notice to merchants to inform about UPE.
+ *
+ * @package WooCommerce\Payments\Admin
+ */
+
+use Automattic\WooCommerce\Admin\Notes\NoteTraits;
+use Automattic\WooCommerce\Admin\Notes\Note;
+use Automattic\WooCommerce\Admin\Notes\WC_Admin_Note;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class WC_Stripe_UPE_Availability_Note
+ */
+class WC_Stripe_UPE_Availability_Note {
+	use NoteTraits;
+
+	/**
+	 * Name of the note for use in the database.
+	 */
+	const NOTE_NAME = 'wc-stripe-upe-availability-note';
+
+	/**
+	 * Link to enable the UPE in store.
+	 */
+	const ENABLE_IN_STORE_LINK = '?page=wc_stripe-onboarding_wizard';
+
+
+	/**
+	 * Get the note.
+	 */
+	public static function get_note() {
+		$note_class = self::get_note_class();
+		$note       = new $note_class();
+
+		$note->set_title( __( 'Boost your sales with the new payment experience in Stripe', 'woocommerce-gateway-stripe' ) );
+		$note->set_content( __( 'Get early access to an improved checkout experience, now available to select merchants. <a href="?TODO" target="_blank">Learn more</a>.', 'woocommerce-gateway-stripe' ) );
+		$note->set_type( $note_class::E_WC_ADMIN_NOTE_INFORMATIONAL );
+		$note->set_name( self::NOTE_NAME );
+		$note->set_source( 'woocommerce-gateway-stripe' );
+		$note->add_action(
+			self::NOTE_NAME,
+			__( 'Enable in your store', 'woocommerce-gateway-stripe' ),
+			self::ENABLE_IN_STORE_LINK,
+			$note_class::E_WC_ADMIN_NOTE_UNACTIONED,
+			true
+		);
+
+		return $note;
+	}
+
+	/**
+	 * Get the class type to be used for the note.
+	 *
+	 * @return string
+	 */
+	private static function get_note_class() {
+		if ( class_exists( 'Automattic\WooCommerce\Admin\Notes\Note' ) ) {
+			return Note::class;
+		} else {
+			return WC_Admin_Note::class;
+		}
+	}
+
+	public static function init() {
+		if ( defined( 'WC_VERSION' ) && version_compare( WC_VERSION, '4.4.0', '>=' ) ) {
+			self::possibly_add_note();
+		}
+	}
```

Comment:
This version check exists because older versions of WC don't include a `NoteTraits` trait.

So, if this file is included on older versions of WC, there would be a fatal error saying that `NoteTraits` doesn't exist, even before this check is executed.

I'd recommend placing this check _before_ the inclusion of the file (and instantiation of the class), and maybe moving the checks in `create_upe_availability_note` into this `init` static method.

Reply:
Thanks for the suggestions. 

The check is already present before the inclusion of the file [here](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/2251ac7f417806fbe73843bf8787dfca1bc73152/woocommerce-gateway-stripe.php#L214). (This issue with `NoteTraits` was caught in the compatibility check for version 3.0.0 and then fixed it in a later commit)

I have addressed the second feedback about moving the checks inside the init method in [this commit](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1762/commits/3f9bdca1e2a86d1b19b6ecf5fcfbc944e6860a39). Also removed the unnecessary duplicate check for version as it is already checked where the file is included.